curvefs/src/metaserver/inode_manager.cpp

```diff
@@ -183,6 +197,86 @@ MetaStatusCode InodeManager::UpdateInode(const Inode &inode) {
     return MetaStatusCode::OK;
 }
 
+MetaStatusCode InodeManager::UpdateInodeWhenCreateSubNode(uint32_t fsId,
+    uint64_t inodeId) {
+    VLOG(1) << "UpdateInodeWhenCreateSubNode, fsId = " << fsId
+            << ", inodeId = " << inodeId;
+    NameLockGuard lg(inodeLock_, GetInodeLockName(fsId, inodeId));
+
+    Inode inode;
+    MetaStatusCode ret = inodeStorage_->Get(
+        InodeKey(fsId, inodeId), &inode);
+    if (ret != MetaStatusCode::OK) {
+        LOG(ERROR) << "GetInode fail, " << inode.ShortDebugString()
+                   << ", ret = " << MetaStatusCode_Name(ret);
+        return ret;
+    }
+    uint32_t oldNlink = inode.nlink();
+    if (oldNlink == 0) {
+        // already be deleted
+        return MetaStatusCode::OK;
+    }
+    inode.set_nlink(++oldNlink);
+
+    struct timespec now;
+    clock_gettime(CLOCK_REALTIME, &now);
+    inode.set_ctime(now.tv_sec);
+    inode.set_ctime_ns(now.tv_nsec);
+    inode.set_mtime(now.tv_sec);
+    inode.set_mtime_ns(now.tv_nsec);
+
+    ret = inodeStorage_->Update(inode);
+    if (ret != MetaStatusCode::OK) {
+        LOG(ERROR) << "UpdateInode fail, " << inode.ShortDebugString()
+                   << ", ret = " << MetaStatusCode_Name(ret);
+        return ret;
+    }
+
+    VLOG(1) << "UpdateInodeWhenCreateSubNode success, "
+            << inode.ShortDebugString();
+    return MetaStatusCode::OK;
+}
+
+MetaStatusCode InodeManager::UpdateInodeWhenRemoveSubNode(uint32_t fsId,
```

Comment:
it seems these two functions only differ in update inode nlink, previous one is increasing and this one is decreasing.

Reply:
done